### PR TITLE
Add visitor class to Associative ASTs

### PR DESCRIPTION
**Pull request by ke-yu**
This PR removes some dead Associaitve AST nodes: `BreakNode`, `ContinueNode`, `ForLoopNode` and `PostFixNode`, and removes them from parser as well.

During the implementation of node to code, it is quite frequently to go through ASTs to do some replacment, for example, to replace an identifier "a" to "b", so two visitor classes are added to Associative AST to support AST traversal and modification.  Unit test `AstVisitorTest.TestAstVisitorForReplacement` shows how to use these classes. 

In the long term, we may support more code analysis functionalists like query or re-factoring. It is too cumbersome to do DFS traversal again and again,

@junmendoza , @lukechurch  please take a look.

### CODE REVIEW

File: src/Engine/ProtoCore/SyntaxAnalysis/AstVisitor.cs [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+        }
+
+        public virtual void Visit(Node node)
+        {
+            AssociativeNode assocNode = node as AssociativeNode;
+            if (assocNode != null)
+                assocNode.Accept(this);
+        }
+
+        public virtual void VisitCommentNode(CommentNode node)
```
Comment (junmendoza) [inline src/Engine/ProtoCore/SyntaxAnalysis/AstVisitor.cs]: Ill need to brush up on visitor patterns, but does this mean that each ASTNode must implement its own DefaultVisit? or Visit function?

File: src/Engine/ProtoCore/SyntaxAnalysis/AstVisitor.cs [hunk 2]
```diff
@@ -0,0 +1,400 @@
+﻿using System;
```
Comment (junmendoza) [inline src/Engine/ProtoCore/SyntaxAnalysis/AstVisitor.cs]: For now, will it only be N2C that will use ASTVisitor?

File: src/Engine/ProtoCore/SyntaxAnalysis/AstVisitor.cs (hunk 1 above)
Comment (ke-yu) [inline src/Engine/ProtoCore/SyntaxAnalysis/AstVisitor.cs]: Yep, all Associaitve AST node overrides `Accept()` method.

File: src/Engine/ProtoCore/SyntaxAnalysis/AstVisitor.cs (hunk 2 above)
Comment (ke-yu) [inline src/Engine/ProtoCore/SyntaxAnalysis/AstVisitor.cs]: Not only N2C. For example, we do identifier mapping in code block node, that mapping could use visitor pattern.

**Comment by junmendoza** [comment]
Ok for me moving forwards. LGTM

**Pull request merged**